packages/next/server/web/sandbox/context.ts

```diff
@@ -266,7 +266,9 @@ function buildEnvironmentVariablesFrom(
   keys: string[]
 ): Record<string, string | undefined> {
   const pairs = keys.map((key) => [key, process.env[key]])
-  return Object.fromEntries(pairs)
+  const env = Object.fromEntries(pairs)
+  env.__NEXT_RUNTIME = 'edge'
```

Comment:
Do we need to always expose this or can we only expose it if it's referenced like other environment variables in the edge runtime?

Reply:
It's used inside the next server code to determine wether we use node or polyfill dependencies, so it's required at the very beginning that can't remove atm. I'll keep it in mind to see if we can opt it in the later versions